framework/src/userobject/ActivateElementsUserObjectBase.C

```diff
@@ -0,0 +1,463 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ActivateElementsUserObjectBase.h"
+#include "DisplacedProblem.h"
+
+#include "libmesh/quadrature.h"
+#include "libmesh/parallel_algebra.h"
+#include "libmesh/parallel.h"
+#include "libmesh/point.h"
+#include "libmesh/dof_map.h"
+
+#include "libmesh/parallel_ghost_sync.h"
+#include "libmesh/mesh_communication.h"
+
+InputParameters
+ActivateElementsUserObjectBase::validParams()
+{
+  InputParameters params = ElementUserObject::validParams();
+  params.addClassDescription("Determine activated elements.");
+  params.addRequiredParam<subdomain_id_type>("active_subdomain_id", "The active subdomain ID.");
+  params.addParam<subdomain_id_type>(
+      "inactive_subdomain_id",
+      -1,
```

Comment:
`subdomain_id_type` is an unsigned integer type. Better use `libMesh::invalid_uint` here!